src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantDiagnosticsWhenPackageCannotBeFound.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Text.RegularExpressions;
+using FluentAssertions;
+using Microsoft.NET.Build.Tasks;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantDiagnosticsWhenPackageCannotBeFound : SdkTest
+    {
+        public GivenThatWeWantDiagnosticsWhenPackageCannotBeFound(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [Fact]
+        public void It_reports_missing_package_deleted_since_restore()
+        {
+            var package = GeneratePackageToGoMissing();
+
+            var project = new TestProject
+            {
+                Name = "MissingPackageDeletedSinceRestore",
+                TargetFrameworks = "netstandard1.3",
+                IsSdkProject = true,
+                PackageReferences = { package },
+            };
+
+            project.AdditionalProperties.Add(
+                "RestoreAdditionalProjectSources",
+                Path.GetDirectoryName(package.NupkgPath));
+
+            var asset = _testAssetsManager
+                .CreateTestProject(project, project.Name)
+                .Restore(Log, project.Name);
+
+            RemovePackageFromCache(package);
+
+            var build = new BuildCommand(
+                Log,
+                Path.Combine(asset.TestRoot, project.Name));
+
+            build.Execute()
+                 .Should()
+                 .Fail()
+                 .And.HaveStdOutContaining(package.ID)
+                 .And.HaveStdOutContaining(package.Version)
+                 .And.NotHaveStdOutContaining("MSB4018"); // unhandled task exception
+
+            // check that incremental build succeeds after a second restore to put back the package
+            asset.Restore(Log, project.Name);
+            build.Execute()
+                 .Should()
+                 .Pass();
+        }
+
+        private static void RemovePackageFromCache(TestPackageReference package)
+        {
+            var shaFile = Path.Combine(
+               TestContext.Current.NuGetCachePath,
+               package.ID,
+               package.Version,
+               $"{package.ID}.{package.Version}.nupkg.sha512");
```

Comment:
Add comment here explaining why deleting the SHA works here.

Reply:
Done in 315e27a